src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionRepeatDbHandler.java

```diff
@@ -0,0 +1,211 @@
+
+	static Logger log = (Logger) LoggerFactory.getLogger(SubscriptionRepeatDbHandler.class);
+
+	
+    @Autowired
+    public MongoDBHandler mongoDbHandler;
+    
+    ObjectMapper mapper = new ObjectMapper();
+    
+    @Getter
+    @Setter
+    @Value("${database.name}")
+    public String dataBaseName;
+    @Getter
+    @Setter
+    @Value("${subscription.collection.repeatFlagHandlerName}")
+    public String collectionName;
+    
+    
+    /*
+     * RepeatFlagHandling structure in MongoDb:
+     * {
+         "_id" : ObjectId("5ac62b4ea4f87e29e8cc5915"),
+         "subscriptionId" : "subsA",
+         //               RequirementId is Requirement List index number from Subscription json object requirement field/list.
+                     <RequirementId>   <AggrObjIds>
+         "requirement" : ["0" : [      
+                      	               "11112", 
+                                       "72324", 
+                                       "72364", 
+                      				   "72233", 
+                                       "71233"
+                                ],
+                          "1" : [ 
+                      	               "11112", 
+                                       "72324", 
+                                       "72364", 
+                      				   "72233", 
+                                       "71233"
+                                ]
+                         ]
+       }
+     * 
+     */
+    
+    /*
+     * Function that stores the matched aggregatedObjectId to the database.
+     * 
+     */
+    public void addMatchedAggrObjToSubscriptionId(String subscriptionId, Integer requirementId, String aggrObjId) throws Exception {
+    	
+    	log.debug("Adding/Updating matched AggrObjId: " + aggrObjId + " to SubscriptionsId: " + subscriptionId + " aggrId matched list" );
+    	
+    	if (checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(subscriptionId, requirementId, aggrObjId)) {
+    		log.info("Subscription: " + subscriptionId + " and AggrObjId, " +
+    							aggrObjId + " has already been matched." +
+    							"No need to register the subscription match.");		
+    		return;
+    	}
+    	
+    	String subscriptionQuery = "{\"subscriptionId\" : \"" + subscriptionId + "\"}";
+    	ArrayList<String> objArray = mongoDbHandler.find(dataBaseName, collectionName, subscriptionQuery);
+    	if (objArray != null && !objArray.isEmpty()) {
+        	BasicDBObject subsObj = (BasicDBObject)  JSON.parse(objArray.get(0));
+
+    		log.debug("SubscriptionId found in Db: " + subscriptionId);
+    		log.debug("SubscriptionIds document content: " + subsObj.toString());
+    		 
+    		
+			JsonNode jNode = null;
+			try {
+				jNode = mapper.readTree(subsObj.get("requirements").toString());
+			} catch (JsonProcessingException e) {
+				e.printStackTrace();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			Iterator<JsonNode> aggrIterator = jNode.get(requirementId).elements();
```

Comment:
This code needs to be in the try space. jnode might be null here if an exception occurs.